### DISCUSSION TITLE
transport resQueue

### DISCUSSION THREAD

**Discussion by danielhrisca** [christoph2/pyxcp#154] [General]
Hello Christoph,

why do we use `BaseTransport.resQueue` as well as `BaseTransport.policy.resQueue`? 

`BaseTransport.resQueue` is consumed by the `get` function, but `BaseTransport.policy.resQueue` is always increasing

https://github.com/christoph2/pyxcp/blob/60a38cbea0b1335d205d1655a4c594fcc424557b/pyxcp/transport/base.py#L373

**Comment by christoph2** [comment]
Well, `BaseTransport.resQueue` is obviously needed for internal purposes, and yes, raw XCP responses are only of interest if one is writing a trace/analyze, or similar application.
The whole idea is, let the user decide how to handle which frames.
For this reason, each each policy has a `filter_out`paramter (`include` and `exclude` are probably more useful), yes, the policy handling is currently a bit cumbersome, but may improved by the coming-soon traitlets based configuration system.
